lib/collection.js

```diff
@@ -1515,6 +1515,13 @@ Collection.prototype.findOneAndReplace = function(filter, replacement, options,
   if (replacement == null || typeof replacement !== 'object')
     throw toError('replacement parameter must be an object');
 
+  // Check that there are no atomic operators
+  const keys = Object.keys(replacement);
+
+  if (keys[0][0] === '$') {
```

Comment:
if `keys[0][0]` guaranteed to always exist? Won't this throw an exception if someone accidentally passes in an empty object for `replacement`?

Reply:
Good call